pkg/parser/base.go

```diff
@@ -0,0 +1,60 @@
+package parser
+
+import (
+	"github.com/k8snetworkplumbingwg/linuxptp-daemon/pkg/config"
+	ptpMetrics "github.com/k8snetworkplumbingwg/linuxptp-daemon/pkg/parser/metrics"
+	"github.com/k8snetworkplumbingwg/linuxptp-daemon/pkg/parser/shared"
+	"strings"
+)
+
+type MetricsExtractor interface {
+	ProcessName() string
+	Extract(configName, logLine string, ifaces config.IFaces) (error, *Metrics)
+}
+
+type BaseMetricsExtractor struct {
+	ProcessNameStr   string
+	ExtractSummaryFn func(messageTag, configName, logLine string, ifaces config.IFaces, _ shared.SharedState) (error, *Metrics)
+	ExtractRegularFn func(messageTag, configName, logLine string, ifaces config.IFaces, _ shared.SharedState) (error, *Metrics)
```

Comment:
Do we need separate functions? for these or can we inspect the log line and dispatch there?